tests/ui_tools/test_boxes.py

```diff
@@ -709,7 +737,8 @@ def test_keypress_CYCLE_COMPOSE_FOCUS(self, write_box, tab_key,
                 write_box.stream_box_view(stream_id)
         else:
             write_box.private_box_view()
-        size = widget_size(write_box)
+        # FIXME: Use widget_size(), instead of hard-coded size.
```

Comment:
Why aren't we using `widget_size`?

Reply:
Well, I tried using `widget_size`, but for some cases, the `write-box` changes to box widget instead of flow and returns a `(maxcol, maxrow)` size instead of just `(maxcol, )`. This made the tests fail for such cases. Let me try to dig more into this and see if I can get a fix for this.

Reply:
Sorry for the late reply. I got busy with other PR's since finding the root+fix of this particular issue was taking forever. :(
I have tried to dig more on this, but it seems like it's an internal `urwid` issue that I am unable to resolve. While looking at this, I realized that the whole `write_box` is sometimes recognized as "flow widget" for all tests (that's when all the tests pass), but sometimes the `write_box` is determined as "box widget" for all tests (2 tests fail in this case).

I also found that in the latter case, the tests only fail for "DOWN" keypresses, because it can't be handled properly by `super()` i.e. urwid's `container.py`. I can fix the whole issue by not letting the "DOWN" keypress being handled by urwid and returning early after manually handling, but it seems to have other side-effects and does not look clean.